WorkspaceServer.Tests/Kernel/KernelExtensionLoaderTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System.Threading.Tasks;
+using Xunit;
+using System.IO;
+using WorkspaceServer.Tests.Kernel;
+using Xunit.Abstractions;
+using MLS.Agent.Tools;
+using Microsoft.DotNet.Interactive;
+using FluentAssertions;
+using Microsoft.DotNet.Interactive.Events;
+using System.Linq;
+using MLS.Agent.Tools.Tests;
+using Microsoft.DotNet.Interactive.Commands;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace WorkspaceServer.Tests
+{
+    public class KernelExtensionLoaderTests : CSharpKernelTestBase
+    {
+        public KernelExtensionLoaderTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task Can_load_from_assembly()
+        {
+            var directory = Create.EmptyWorkspace().Directory;
+            var extensionOutputDirectory = directory.CreateSubdirectory("extensionOutput");
+            var extensionDll = await CreateExtensionDll(directory, extensionOutputDirectory);
+
+            var kernel = CreateKernel();
+            await new KernelExtensionLoader().TryLoadFromAssembly(extensionDll, kernel);
+
+            KernelEvents.Should()
+                      .ContainSingle(e => e.Value is CodeSubmissionEvaluated &&
+                                          e.Value.As<CodeSubmissionEvaluated>().Code.Contains("using System.Reflection;"));
+
+        }
+
+        private static async Task<FileInfo> CreateExtensionDll(DirectoryInfo baseDirectory, DirectoryInfo extensionOutputDirectory)
+        {
+            var extensionCodeDirectory = baseDirectory.CreateSubdirectory("extensionCode");
+            var microsoftDotNetInteractiveDllPath = typeof(IKernelExtension).Assembly.Location;
+            var extensionName = baseDirectory.Name;
+
+            new InMemoryDirectoryAccessor(extensionCodeDirectory)
```

Comment:
Please refactor so we don't have multiple copies of this code.

Reply:
Done. Thanks!